### DISCUSSION TITLE
`forwardRef` error when dynamically importing a module

### DISCUSSION THREAD

**Discussion by matiastucci** [vercel/next.js#16309] [Help]
I'm trying to use `forwardRef` on an external component but `ref.current` is not the actual `ref`. Not sure if I'm missing something.

I'm doing:

```js
const Editor = dynamic(() => import("react-markdown-editor-lite"), {
  ssr: false
});

const ForwardedRefEditor = forwardRef((props, ref) => (
  <Editor {...props} ref={ref} />
));

-----

const editorRef = useRef(null);
<ForwardedRefEditor
  ref={editorRef}
  value={content}
  onChange={({ text }) => setContent(text)}
/>
```

Full code: https://codesandbox.io/s/objective-benz-qh4ec?file=/pages/index.js:63-73

More info: https://github.com/vercel/next.js/issues/4957

**Comment by matiastucci** [comment]
I'm still trying to fix this. Any ideas?

**Comment by parkjoon** [comment]
hey can you try this for me:

```
const Editor = dynamic(() => import("react-markdown-editor-lite"), {
  ssr: false
});

const ForwardedRefEditor = forwardRef((props, ref) => (
  <Editor {...props} ref={el => ref = el} /> // <- this is the only line i changed
));

-----

const editorRef = useRef(null);
<ForwardedRefEditor
  ref={editorRef}
  value={content}
  onChange={({ text }) => setContent(text)}
/>
```
It MIGHT work if you set ref as a function. I had a similar problem and this change worked for me.

If it works can you let me  know :)

**Reply by matiastucci** [reply]
hi! I just tried that and `editorRef.current` is `null` 😞

**Comment by apecollector** [comment]
I'm hitting the same issue

**Reply by geminiyellow** [reply]
the error is still happen

**Comment by 1hachem** [comment]
getting the same error
